### PR TITLE
mcs: refactor refill_budget_check to ease verification

### PR DESCRIPTION
**Pull request by michaelmcinerney**
This refactors refill_budget_check and schedule_used to ease verification by limiting heap access via refill_head and refill_tail, and providing (inlined) named functions for both the condition and bodies of the loops.

### CODE REVIEW

**Comment by michaelmcinerney** [comment]
Force pushed with a shorter commit message, to stop Gitlint complaining

**Review by Indanz — APPROVED** [review]
The diff is a bit confusing, but I'm now convinced it's functionally the same.

File: src/kernel/sporadic.c [hunk 1]
```diff
@@ -247,46 +248,83 @@ static bool_t refill_head_overlapping(sched_context_t *sc)
...
+    /* Delay head to ensure the subsequent refill doesn't end any
+     * later (rather than simply combining refills). */
+    refill_head(sc)->rTime -= head.rAmount;
+}
+
+void refill_budget_check(ticks_t usage)
+{
+    sched_context_t *sc = NODE_STATE(ksCurSC);
+    assert(!isRoundRobin(sc));
+    REFILL_SANITY_START(sc);
```
Comment (Xaphiosis) [inline src/kernel/sporadic.c:311]: out of curiosity, what does this do? is there a corresponding `REFILL_SANITY_START`?

**Review by Xaphiosis — APPROVED** [review]

File: src/kernel/sporadic.c (hunk 1 above)
Comment (Indanz) [reply src/kernel/sporadic.c:311]: There's a REFILL_SANITY_END at the bottom. It just checks that the refill_sum didn't change and that refills are ordered and the SC is still round-robin.

**Review by lsf37 — APPROVED** [review]

**Pull request merged**